### PR TITLE
[Fixes #1510] UI, Critical: retention.bytes parameter set to 0

### LINKED ISSUE #1510: UI, Critical: retention.bytes parameter set to 0
**Describe the bug**

Go to topic creation view -> enter new topic name, do not edit other configs  ->  retention.bytes properties are set to 0 in passed parameter.

Expected:  when `Max size on disk in GB` is "Not set"`retention.bytes` should be either not passed or passed as -1 which indicates that topic don't have size limit.

This is critical because it will cause unexpected topic data deletion, need to fix it ASAP

Screenshot 2022-01-28 at 18 32 02

### PR DESCRIPTION
**Pull request by Motherships**
- [ ] **Breaking change?** (if so, please describe the impact and migration path for existing applications:)

**What changes did you make?** (Give an overview)
- Fixes #1510 

**Is there anything you'd like reviewers to focus on?**
You can test it manually, I did it myself, I think it works correctly

image

**How Has This Been Tested?** (put an "X" next to an item)

- [X] No need to
- [ ] Manually(please, describe, when necessary)
- [ ] Unit checks
- [ ] Integration checks
- [ ] Covered by existing automation

**Checklist** (put an "X" next to an item, otherwise PR will fail)
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (e.g. **ENVIRONMENT VARIABLES**)
- [X] My changes generate no new warnings(e.g. Sonar is happy)
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged

Check out [Contributing](https://github.com/provectus/kafka-ui/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/provectus/kafka-ui/blob/master/CODE-OF-CONDUCT.md)

### CODE REVIEW

**Review by Haarolean — CHANGES_REQUESTED** [review]
Can we add a test so this doesn't happen again?

**Review by workshur — APPROVED** [review]
Agree about tests

**Review by NelyDavtyan — COMMENTED** [review]
Request review again when you will be finished with tests please

**Comment by Motherships** [comment]
Changes from previous commit:
- Adds test for TopicForm
- Removes bug from Topic/New test 
- Updates Topic/New test
- Adds accessible names for TopicForm
- Add id's to Select component

And I don't really want to make changes proposed by Sonar as I think they are won't make code better
It's just generic testing library `expect`'s with querying by role and accessible name

**Comment by Motherships** [comment]
@Haarolean said that we can't mute duplication error from sonar - so i will rewrite this part

**Review by NelyDavtyan — APPROVED** [review]

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      userEvent.type(screen.getByPlaceholderText('Topic Name'), topicName);
+      userEvent.click(screen.getByText('Send'));
+    });
+
+    await waitFor(() =>
+      expect(mockedHistory.location.pathname).toBe(
+        clusterTopicPath(clusterName, topicName)
+      )
+    );
+    await waitFor(() => expect(mockedHistory.push).toBeCalledTimes(1));
```
Comment (workshur) [inline kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:87]: I'm not sure we need waitFor here. You have been waiting this on a previous line

**Review by workshur — APPROVED** [review]

File: kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx (hunk 1 above)
Comment (Motherships) [reply kafka-ui-react-app/src/components/Topics/New/__test__/New.spec.tsx:87]: Y, it seems after we done waiting for `pathname` change all other mocks are already changed and we don't need to wait for them
thanks
image

**Pull request merged**